docs/book/src/reference/creating-events.md

```diff
@@ -0,0 +1,109 @@
+# Creating Events
+
+It is often useful to publish *Event* objects from the controller Reconcile function. 
+
+Events allow users to see what is going on with a particular object, and allow automated processes to see and respond to them.
+
+ Recent Events for an object can be viewed by running `kubectl describe <resource kind> <resource name>`.
+ 
+  ### how to implement the solution to raise an event:
+
+a) You need to pass the recorder when you set up the recorder for the controller when the event will be raised see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/main.go#L103
+b) You need to add the makers to add the RBAC permissions to allow your Operator/controller to raise the event see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/controllers/memcached_controller.go#L66 and run make manifests
```

Comment:
````suggestion
### Passing the EventRecorder to the Controller

Now, we will pass the EventRecorder when we are setting up the controller on the Manager. See that we will change the implementation scaffolded in `cmd/main.go`:

```go
	if err = (&controller.MyKindReconciler{
		Client:   mgr.GetClient(),
		Scheme:   mgr.GetScheme(),
		// Note that we added the following line: 
		Recorder: mgr.GetEventRecorderFor("mykind-controller"),
	}).SetupWithManager(mgr); err != nil {
		setupLog.Error(err, "unable to create controller", "controller", "MyKind")
		os.Exit(1)
	}
```

````

Reply:
Now, that we changed the controller, then we can describe how to pass the struct to it.